nautobot/utilities/renderers.py

```diff
@@ -1,15 +1,169 @@
+                    else:
+                        pk_list = model.objects.values_list("pk", flat=True)
+                else:
+                    form_cls = view.get_form()
+                    pk_list = request.POST.getlist("pk")
+                    # form = form_cls(request.POST)
+                    form = form_cls(
+                        initial={
+                            "pk": pk_list,
+                            "return_url": view.get_return_url(request),
+                        }
+                    )
+            elif view.action == "bulk_create":
+                form = view._import_form_for_bulk_import()
+            elif view.action == "bulk_edit":
+                if request.POST.get("_all") and view.bulk_edit_filterset is not None:
+                    pk_list = [obj.pk for obj in view.bulk_edit_filterset(request.GET, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+                if "_apply" in request.POST:
+                    form = view.bulk_edit_form(model, request.POST)
+                    restrict_form_fields(form, request.user)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    # Check for other contextual data needed for the form. We avoid passing all of request.GET because the
+                    # filter values will conflict with the bulk edit form fields.
+                    # TODO: Find a better way to accomplish this
+                    if "device" in request.GET:
+                        initial_data["device"] = request.GET.get("device")
+                    elif "device_type" in request.GET:
+                        initial_data["device_type"] = request.GET.get("device_type")
+
+                    form = view.bulk_edit_form(model, initial=initial_data)
+                    restrict_form_fields(form, request.user)
+
+        context = {
+            "obj": obj,
+            "object": instance,
+            "obj_type": view.queryset.model._meta.verbose_name,
+            "obj_type_plural": view.queryset.model._meta.verbose_name_plural,
+            "editing": obj.present_in_database,
+            "form": form,
+            "fields": view.import_form().fields,
+            "table": table if table else data.get("table", None),
+            "return_url": view.get_return_url(request, instance),
+            "verbose_name": view.queryset.model._meta.verbose_name,
+            "verbose_name_plural": view.queryset.model._meta.verbose_name_plural,
+            "changelog_url": view.get_changelog_url(instance),
+            "content_type": content_type,
+            "permissions": permissions,
+            "action_buttons": valid_actions,
+            "table_config_form": TableConfigForm(table=table) if table else None,
+            "filter_form": view.filterset_form(request.GET, label_suffix="") if view.filterset_form else None,
```

Comment:
`view.filterset_form_class`?

And maybe just to keep it clean could we have another helper that does `get_filter_form()` or something like that to encapsulate this logic and support overloading?